scylla/src/authentication/mod.rs

```diff
@@ -1,23 +1,27 @@
-use async_trait::async_trait;
 use bytes::{BufMut, BytesMut};
 
+use crate::utils::futures::BoxedFuture;
+
 /// Type to represent an authentication error message.
 pub type AuthError = String;
 
+/// Type to represent an initial auth response with an authenticator session.
+pub(crate) type AuthInitialResponseAndSession = (Option<Vec<u8>>, Box<dyn AuthenticatorSession>);
```

Comment:
This is another case of the same problem of `pub(crate)` in `pub` item.
How does it look in generated documentation?

Reply:
Yeah, it's the same as for `BoxedFuture` alias. There is no problem in the docs:
![image](https://github.com/scylladb/scylla-rust-driver/assets/64713071/2788af7b-39fc-4f2a-9cc3-e92409c68da7)

However, when I use the driver in some user application and generate the definition using `vscode` hints, the generated function signature contains the opaque type:
```rust
struct Foo;

impl AuthenticatorProvider for Foo {
    fn start_authentication_session<'a>(
        &'a self,
        authenticator_name: &'a str,
    ) -> scylla::BoxedFuture<
        '_,
        std::prelude::v1::Result<AuthInitialResponseAndSession, scylla::authentication::AuthError>,
    > {
        todo!()
    }
}
```

What's interesting tho, is that the compiler can deduce the type in the error message when the function is not implemented:
```
not all trait items implemented, missing: `start_authentication_session`
  --> src/main.rs:14:1
   |
14 | impl AuthenticatorProvider for Foo {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `start_authentication_session` in implementation
   |
   = help: implement the missing item: `fn start_authentication_session(&'a self, _: &'a str) -> Pin<Box<(dyn Future<Output = std::result::Result<(Option<Vec<u8>>, Box<(dyn AuthenticatorSession + 'static)>), String>> + Send + 'a)>> { todo!() }`
```

I'm not sure why vscode generates the function declaration with an opaque type.

I think that we should make this alias public as well. However, i think we should come up with a more fitting name for this type alias (or introduce a struct instead). I only introduced this alias because clippy was complaining about the complexity of the returned type and I wanted the alias to be used internally - unfortunately, it appears in the public trait declaration.